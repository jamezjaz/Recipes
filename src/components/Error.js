import React from 'react';

const Error = () => (
  <div>
    <h1 style={{ color: 'red' }}>Oops! Page not found!</h1>
  </div>
);

export default Error;
