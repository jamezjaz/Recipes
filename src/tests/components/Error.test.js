import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import Error from '../../components/Error';

describe('<Error />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Error />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('contains a header text', () => {
    render(
      <Router>
        <Error />
      </Router>,
    );
    expect(screen.getByText('Oops! Page not found!')).toBeInTheDocument();
  });
});
