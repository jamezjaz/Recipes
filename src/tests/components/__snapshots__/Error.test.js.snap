// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Error /> renders correctly 1`] = `
<div>
  <h1
    style={
      Object {
        "color": "red",
      }
    }
  >
    Oops! Page not found!
  </h1>
</div>
`;
